qiskit/circuit/add_control.py

```diff
@@ -97,44 +97,35 @@ def control(operation: Union[Gate, ControlledGate],
     q_ancillae = None  # TODO: add
     qc = QuantumCircuit(q_control, q_target)
 
-    if operation.name == 'x' or (
+    if isinstance(operation, XGate) or (
             isinstance(operation, controlledgate.ControlledGate) and
-            operation.base_gate.name == 'x'):
+            isinstance(operation.base_gate, XGate)):
```

Comment:
I was going to add a case for cx below, but realized it's already handled here for arbitrary controls. Should we do the same for the other gate checks below?

Reply:
This might be good to do for the rotation gates as well. It would be interesting to check if it reduces gate count.

Reply:
@ewinston gave this a shot in the commit below, but it's breaking add_control for RZ and CRZ gates. Not sure where to start investigating... any ideas where to dig further?

Reply:
I'm not sure why this breaks but if you remove the `or` condition in `_operation_has_base-gate` the issue seems to resolve. Then it's just checkout `isinstance(operation, gate)` like in original logic.